### PR TITLE
Test XLA examples

### PR DESCRIPTION
**Pull request by LysandreJik**
Add a script to test examples on XLA.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -73,7 +73,7 @@ jobs:
             - checkout
             - run: sudo pip install .[sklearn,torch,testing]
             - run: sudo pip install -r examples/requirements.txt
-            - run: python -m pytest -n 8 --dist=loadfile -s ./examples/ | tee output.txt
+            - run: python -m pytest -n 8 --dist=loadfile -s ./examples/ --ignore=examples/test_xla_examples.py | tee output.txt
```
Comment (sshleifer) [inline .circleci/config.yml:76]: you could also make a `require_xla` decorator.

File: examples/test_xla_examples.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+                for line in lines:
+                    key, value = line.split(" = ")
+                    result[key] = float(value)
+
+            del result["eval_loss"]
+            for value in result.values():
+                # Assert that the model trains
+                self.assertGreaterEqual(value, 0.70)
+
+            # Assert that the script takes less than 100 seconds to make sure it doesn't hang.
```
Comment (sshleifer) [inline examples/test_xla_examples.py:87]: (nit) You can do this with pytimeout

**Review by sshleifer — APPROVED** [review]
Love this!

File: .circleci/config.yml (hunk 1 above)
Comment (LysandreJik) [reply .circleci/config.yml:76]: Very good idea!

File: examples/test_xla_examples.py (hunk 2 above)
Comment (LysandreJik) [reply examples/test_xla_examples.py:87]: Nice, will do!

**Pull request merged**